crates/nu-engine/src/eval_ir.rs

```diff
@@ -1270,6 +1273,20 @@ fn check_type(val: &Value, ty: &Type) -> Result<(), ShellError> {
     }
 }
 
+/// Type check and convert value for assignment.
+fn convert_value_for_assignment(val: Value, target_ty: &Type) -> Result<Value, ShellError> {
```

Comment:
I don't think "convert" is really accurate anymore since it's really just doing type checking now. Could you rename this function to something like `check_assignment_type` and update the comment?